packages/teleport-project-generator-react-basic/src/utils.ts

```diff
@@ -1,18 +1,37 @@
-import { createFolder } from '@teleporthq/teleport-generator-shared/lib/utils/project-utils'
-
 import {
   GeneratedFile,
   GeneratedFolder,
+  TemplateDefinition,
 } from '@teleporthq/teleport-generator-shared/lib/typings/generators'
+import { injectFilesToPath } from '@teleporthq/teleport-generator-shared/lib/utils/project-utils'
+import {
+  DEFAULT_SRC_FILES_PATH,
+  DEFAULT_COMPONENT_FILES_PATH,
+  DEFAULT_PAGE_FILES_PATH,
+  DEFAULT_STATIC_FILES_PATH,
+} from './constants'
 
 export const buildFolderStructure = (
```

Comment:
can this function be reused for all project generators? I think now we're still relying on the parent-child relationship between the src / pages / components folders but it would be nice to try a generic function based on the template / default structure. let's talk about this maybe